tuist/tuist
# Fix `platformFilter` usage for single filters

Part of  https://github.com/tuist/tuist/pull/5381

### Short description 📝

- XcodeProj format supports both a single `platformFilter` and multiple `platformFilters` (plural) attribute with the project file format
- the singlular attribute appears to only be used for `ios` and `maccatalyst` while for all other filters, the plural form is used (even when specifying a single filter)
- Xcode doesn't recognize the other filters such as `tvos` or `macos` when provided to `platformF ilter`
- Xcode does support `ios` and `maccatlyst` when using the plural form however immediatley revert s it back to `platformFilter` causing an unstable generated project
- To mitigate this those filters are explicitly specifed as the ones that can be used with the si ngle `platformFilter`

### How to test the changes locally 🧐

- platformFilters are not yet exposed via public API as such inspection via manually creating a p roject with a multi-platform target, apply filters and verify which attributes are in use to veri fy

### Contributor checklist ✅


### Reviewer checklist ✅
